### PR TITLE
Uxd 98 html table

### PR DESCRIPTION
**Pull request by nahumzs**
### Purpose 🚀
The Simple Table component provides the functionality of a regular HTML table with a similar API of the DataGrid, unlike the DataGrid the table component lacks certain futures and it's intended to be used when the number of rows is no overwhelming and dynamic height is required.

For cases where the table requires one or more of the following features, the use of the DataGrid can be a better alternative:
- Display hundreds or thousands of rows at the same time
- Fix columns and Fix headers
- Need for up, right, down, left arrow key navigation

Both components DataGrid and Table can be used with the ActionBar without a problem and both of them can have a custom header or custom cells. 

### Notes ✏️
Screen Shot 2020-08-13 at 1 14 35 PM
Screen Shot 2020-08-13 at 1 14 43 PM

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_
- [ ] MINOR (backward compatible) change to _these packages_
- [ ] PATCH (bug fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/UXD-98-HTML-Table

### Screenshots 📸
_optional but highly recommended_

### References 🔗
_relevant Jira ticket / GitHub issues_

### CODE REVIEW

**Comment by midoritakeuchi** [comment]
Hi @nahumzs! Thanks for creating this component for us 👏🏼
One question - In the `DataGrid` component there's the count of rows and columns at the bottom. Will this new `Table` component have the option to hide that? No worries if it is too complicated.

**Comment by nahumzs** [comment]
@midoritakeuchi yes this doesn't have it, actually, the DataGrid shouldn't have it either, but since the DataGrid is intended for displaying a vast amount of rows and columns sometimes make sense to know how many columns and rows are rendered. 

We have a ticket to make the DataGrid Footer Optional in the near future :).

File: packages/Table/src/Table.styles.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    `;
+  }
+  return css`
+    ${zebras};
+  `;
+});
+
+export const Thead = styled.thead`
+  background: ${tokens.table.header.backgroundColor};
+
```
Comment (tristanjasper) [inline packages/Table/src/Table.styles.js:30]: remove empty line

File: packages/Table/stories/Table.collections.stories.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+};
+
+function Title(props) {
+  const { label, description, tag } = props;
+
+  return (
+    <div>
+      <titleStyles.Title>
+        <titleStyles.Label>
+          <Button.Link size="big">{label}</Button.Link>
```
Comment (tristanjasper) [inline packages/Table/stories/Table.collections.stories.js:74]: Shouldn't `size` be a `CustomPropTypes.ShirtSize` value? Do we need to create a new one?

File: packages/Table/stories/Table.collections.stories.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    width: 100%;
+  `,
+};
+
+function Item(props) {
+  const { value } = props;
+
+  return (
+    <itemStyles.Item>
+      <Button kind="flat">{value}</Button>
```
Comment (tristanjasper) [inline packages/Table/stories/Table.collections.stories.js:99]: use `Button.Kinds`? Perhaps not necessary in a story though

File: packages/Table/stories/Table.collections.stories.js (hunk 2 above)
Comment (nahumzs) [reply packages/Table/stories/Table.collections.stories.js:74]: I'm not fan of using micro abstraction, which adds unnecessary complexity IMHO while reading the code. A lot of my thoughts about this are part of the comment that I added about our Constant package :)

Comment (tristanjasper) [reply packages/Table/stories/Table.collections.stories.js:74]: Ok. I figured we'd standardized the approach but maybe not. I'll update the types/enums pr to try and close it out and solidify an approach.

**Comment by oscarkwan** [comment]
Semantic html checklist

- [ ] th are used for headers
- [ ] tr and td are used properly
- [] thead and tbody are used properly

Question
1. Will there be issues with global styling? Or should we just ignore that?
2. According to designs, for the zebra pattern I believe that it should start with `white/grey` and not `grey/white`

File: packages/Table/src/Table.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+  let ColumnDefinitions = extractedColumnDefinitions;
+
+  if (!Array.isArray(extractedColumnDefinitions)) {
+    // when there is only one component extracted function return the element not an array of elements
+    ColumnDefinitions = [extractedColumnDefinitions];
+  }
+
+  return (
+    <sc.Table {...moreProps}>
```
Comment (oscarkwan) [inline packages/Table/src/Table.js:42]: I feel like we should always provide an `aria-label`. Or else screen reader will just say "2 rows 11 columns" and give no context on what the table is

File: packages/Table/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+<!-- autogenerated don't modify -->
+
+# @paprika/table - 0.1.0
+
+## Description
+
+The Table component is a simple wrap for the html table, for more advance feature please take a look into the DataGrid
```
Comment (oscarkwan) [inline packages/Table/README.md:7]:
```suggestion
The Table component is a simple wrap for the html table. For the use of a data table, please take a look into the DataGrid component.
```

File: packages/Table/src/Table.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+import PropTypes from "prop-types";
+import ColumnDefinition from "./components/ColumnDefinition";
+import types from "./types";
+import * as sc from "./Table.styles";
+
+const propTypes = {
+  /** Define the look for borders in the table DataGrid.types.GRID, DataGrid.types.NONE, etc.  */
+  borderType: PropTypes.oneOf(["grid", "non", "horizontal", "vertical"]),
+  children: PropTypes.node.isRequired,
+  /** Add an alternate background on the DataGrid's rows */
```
Comment (oscarkwan) [inline packages/Table/src/Table.js:14]:
```suggestion
  /** Add an alternating background on the table rows */
```

File: packages/Table/src/Table.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+import React from "react";
+
+import extractChildren from "@paprika/helpers/lib/extractChildren";
+import PropTypes from "prop-types";
+import ColumnDefinition from "./components/ColumnDefinition";
+import types from "./types";
+import * as sc from "./Table.styles";
+
+const propTypes = {
+  /** Define the look for borders in the table DataGrid.types.GRID, DataGrid.types.NONE, etc.  */
```
Comment (oscarkwan) [inline packages/Table/src/Table.js:11]:
```suggestion
  /** Define the look for borders in the table Table.types.GRID, Table.types.NONE, etc.  */
```

File: packages/Table/src/Table.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  borderType: PropTypes.oneOf(["grid", "non", "horizontal", "vertical"]),
+  children: PropTypes.node.isRequired,
+  /** Add an alternate background on the DataGrid's rows */
+  hasZebraStripes: PropTypes.bool,
+  /** Array of data to be stored in the DataGrid */
```
Comment (oscarkwan) [inline packages/Table/src/Table.js:16]:
```suggestion
  /** Array of data to be stored in the Table */
```

File: packages/Table/src/Table.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+  data: PropTypes.arrayOf(PropTypes.shape({})),
+  /** Sets the height of the DataGrid */
```
Comment (oscarkwan) [inline packages/Table/src/Table.js:18]: Remove comment

Comment (oscarkwan) [reply packages/Table/src/Table.js:18]: FYI, you will need to update readme after these changes

File: packages/Table/src/Table.js (hunk 4 above)
Comment (nahumzs) [reply packages/Table/src/Table.js:42]: I mean you can totally add it from the consumer side that's why moreProps are there, do you mean we should enforce something like `a11yText` so the consumer always has to provide a label?

Comment (nahumzs) [reply packages/Table/src/Table.js:42]: > According to designs, for the zebra pattern I believe that it should start with white/grey and not grey/white

oh I see good catch will change it.

Comment (oscarkwan) [reply packages/Table/src/Table.js:41]: `a11yText` would be nice i think. what do you think?

**Comment by oscarkwan** [comment]
Reminder to add follow up for tests

**Comment by nahumzs** [comment]
@oscarkwan I don't know what kind of test would we need? I added screener test already, what else do we need for this specific component?

**Comment by oscarkwan** [comment]
> @oscarkwan I don't know what kind of test would we need? I added screener test already, what else do we need for this specific component?

Screener is good, and just some basic unit tests to make sure data passed in is rendered correctly

**Pull request merged**